kong/tools/utils.lua

```diff
@@ -131,7 +131,47 @@ do
     "/etc/ssl/cert.pem",                                 -- OpenBSD, Alpine
   }
 
+  local cert_dirs = {
+    "",
+    "certs",
+  }
+
+  local cert_files = {
+    "ca-certificates.crt",
+    "ca-bundle.crt",
+    "ca-bundle.pem",
+    "cacert.pem",
+    "cert.pem",
+  }
+
   function _M.get_system_trusted_certs_filepath()
+    local version = require("resty.openssl.version")
+    local openssl_dir = version.version(version.DIR)
+
+    if openssl_dir then
+      -- e.g.:
+      -- OPENSSLDIR: "/opt/homebrew/etc/openssl@1.1"
+      openssl_dir = openssl_dir:sub(14, -2)
+      for _, cert_dir in ipairs(cert_dirs) do
+        for _, cert_file in ipairs(cert_files) do
+          local ca_file = pl_path.join(openssl_dir, cert_dir, cert_file)
+          if pl_path.exists(ca_file) then
```

Comment:
this logic will seldomly get picked up because we ships our own openssl library, and its OPENSSL_DIR is not same as the system openssl dir, which usually doesn't contain the ca file.